src/sentry/integrations/opsgenie/actions/notification.py

```diff
@@ -0,0 +1,139 @@
+            )
+            org_integration = integration_service.get_organization_integration(
+                integration_id=integration.id, organization_id=org_integration_id
+            )
+        team = self._get_team(org_integration)
+        if not team:
+            logger.exception(
+                "The Opsgenie team no longer exists, or the team does not belong to the selected account."
+            )
+            return
+
+        def send_notification(event, futures):
+            org_integration = self.get_organization_integration()
+            org_integration_id = None
+            if org_integration:
+                org_integration_id = org_integration.id
+            else:
+                org_integration_id = infer_org_integration(
+                    integration_id=integration.id, ctx_logger=logger
+                )
+            client = OpsgenieClient(
+                integration=integration,
+                org_integration_id=org_integration_id,
+                integration_key=team["integration_key"],
+            )
+            try:
+                rules = [f.rule for f in futures]
+                resp = client.send_notification(event, rules)
+            except ApiError as e:
+                self.logger.info(
+                    "rule.fail.opsgenie_notification",
+                    extra={
+                        "error": str(e),
+                        "team_name": team["team"],
+                        "team_id": team["id"],
+                        "project_id": event.project_id,
+                        "event_id": event.event_id,
+                    },
+                )
+                raise e
+
+            self.logger.info(
+                "rule.success.opsgenie_notification",
+                extra={
+                    "status_code": resp.status_code,
+                    "project_id": event.project_id,
+                    "event_id": event.event_id,
+                    "team_name": team["team"],
+                    "team_id": team["id"],
+                },
+            )
+
+        key = f"opsgenie:{integration.id}:{team['id']}"
+        yield self.future(send_notification, key=key)
+
+    def get_teams(self) -> list[tuple[str, str]]:
+        from sentry.services.hybrid_cloud.integration import integration_service
+
+        organization_integrations = integration_service.get_organization_integrations(
+            providers=[self.provider], organization_id=self.project.organization_id
+        )
+
+        teams = []
+        for oi in organization_integrations:
+            team_table = oi.config.get("team_table")
+            if team_table:
+                teams += [(team["id"], team["team"]) for team in team_table]
+        return teams
+
+    def render_label(self) -> str:
+        team = self._get_team(self.get_organization_integration())
+        if team:
+            team_name = team["team"]
+        else:
+            team_name = "[removed]"
```

Comment:
```suggestion
        team_name = team["team"] if team else "[removed]"
```
nit: i love shortening code